src/main/java/de/filefighter/rest/domain/user/business/UserBusinessService.java

```diff
@@ -74,4 +70,54 @@ public User findUserByUsername(String username) {
 
         return userDtoService.createDto(entity);
     }
+
+    public void registerNewUser(UserRegisterForm newUser) {
+        // check username
+        String username = newUser.getUsername();
+
+        User user = null;
+        try {
+            user = this.findUserByUsername(newUser.getUsername());
+        } catch (UserNotFoundException ignored) {
+        }
+
+        if (null != user)
+            throw new UserAlreadyExistsException("Username already taken.");
+
+        // check pws.
+        String password = newUser.getPassword();
+        passwordIsValid(password);
+
+        String confirmationPassword = newUser.getConfirmationPassword();
+        passwordIsValid(confirmationPassword);
+
+        if (!password.contentEquals(confirmationPassword))
+            throw new UserNotRegisteredException("Passwords do not match.");
+
+        //check groups
+        long[] userGroups = newUser.getGroupIds();
+        if (null == userGroups)
+            userGroups = new long[0];
+
+        for (long id : userGroups) {
+            try {
+                groupRepository.getGroupById(id);
+            } catch (IllegalArgumentException exception) {
+                throw new UserNotRegisteredException("One or more groups do not exist.");
+            }
+        }
+
+        //create new user.
+        userRepository.save(UserEntity.builder()
+                .lowercaseUsername(username.toLowerCase())
+                .username(username)
+                .password(password)
+                .refreshToken(AccessTokenBusinessService.generateRandomTokenValue())
+                .userId(getUserCount() + 1)
```

Comment:
counting up is not perfect, as user Id coud be easily guessed, but could be ok here

Reply:
we will change the ids anyway i guess.